src/email-sender/email-sender.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Example email sending lambda function that works with events.kleros.io.
+ * Replace the email lookup with your subscribers for an event and replace your sendgrid template.
+ */
+const sgMail = require('@sendgrid/mail')
+const Web3 = require('web3')
+
+const dynamoDB = require('../utils/dynamo-db')
+
+module.exports.post = async (event, _context, callback) => {
+  // Get the event body
+  const body = JSON.parse(event.body)
+  if (body == null) {
+    return callback(null, {
+      statusCode: 400,
+      headers: { 'Access-Control-Allow-Origin': '*' },
+      body: JSON.stringify({
+        error: 'No event log passed in body.'
+      })
+    })
+  }
+
+  // FIXME determine which email address you want to use. This example uses the caller of tx that kicks off the event
+  const txHash = body.transactionHash
+
+  // Find the sender of the tx
+  const web3 = new Web3(process.env.PROVIDER_URI)
+  const transaction = await web3.eth.getTransaction(txHash)
+
+  // Fetch from the user-settings table
+  const item = await dynamoDB.getItem({
+    Key: { address: { S: transaction.from } },
+    TableName: 'user-settings',
+    AttributesToGet: ['email']
+  })
+
+  let emailAddress
+  if (item && item.Item && item.Item.email) emailAddress = item.Item.email.S
+
+  if (emailAddress == null)
+    return callback(null, {
```

Comment:
The function essentially is unable to do it's task if there is no email address so I think an error is appropriate. If it is a 204 the server would have no way to know if the email had been sent. The server can choose to ignore the error (the event service does)

Reply:
That's exactly why it's not an error, because it's generic. The server is just calling an event handler, it does not expect an email to be sent or not. All it expects is for the event to be handled appropriately. Not sending the email can be proper handling. `400` is for malformed requests. A request with an event that does not have an email related to it, is not malformed.

Reply:
Yeah, that's a good approach. But, the code should be 200 then, because you have a body now.